packages/analytics/analytics-geo-map/src/components/AnalyticsGeoMap.vue

```diff
@@ -0,0 +1,244 @@
+  if (metric > 60) return '#0D8093'
+  if (metric > 40) return '#009FA9'
+  if (metric > 20) return '#00BDB7'
+  return '#0CDCBD'
+}
+
+// Simplified coords may be 2 or 3 layers
+const flattenPositions = (position: any): number[][] => {
+  const flat: (number | number[])[] = position.flat(1)
+
+  if (Array.isArray(flat[0]) && Array.isArray(flat[0][0])) {
+    return flattenPositions(flat)
+  }
+
+  return flat as number[][]
+}
+
+const goToCountry = (countryCode: string) => {
+
+  // Overrides for large spanning countries
+  if (countryCode === 'RU') {
+    map.value?.fitBounds([
+      [20, 40],
+      [180, 80],
+    ])
+    return
+  } else if (countryCode === 'US') {
+    map.value?.fitBounds([
+      [-130, 20],
+      [-60, 50],
+    ])
+    return
+  } else if (countryCode === 'FR') {
+    map.value?.fitBounds([
+      [-5, 42],
+      [9, 51],
+    ])
+    return
+  } else if (countryCode === 'NO') {
+    map.value?.fitBounds([
+      [4, 57],
+      [32, 71],
+    ])
+    return
+  }
+
+  const found = (props.geoJsonData as FeatureCollection).features.find((f: Feature) => f.properties?.ISO_A2 === countryCode)
+  if (found) {
+    const coordinates = (found.geometry as MultiPolygon)?.coordinates
+    if (!coordinates) return
+    const allCoords = flattenPositions(coordinates)
+
+    const lats = allCoords.map((c: number[]) => c[1])
+    const longs = allCoords.map((c: number[]) => c[0])
+
+    const minLat = Math.min(...lats)
+    const maxLat = Math.max(...lats)
+    const minLong = Math.min(...longs)
+    const maxLong = Math.max(...longs)
+    map.value?.fitBounds([
+      [minLong, minLat],
+      [maxLong, maxLat],
+    ])
+  }
+}
+
+const mapOptions = computed(() => {
+  const options: MapOptions = {
+    container: 'mapContainer',
+    style: { version: 8, sources: {}, layers: [] },
+  }
+  if (!props.center && !props.initialZoom) {
+    options.bounds = [
+      [-180, -90],
+      [180, 90],
+    ]
+  } else {
+    if (props.center) {
+      options.center = props.center as LngLatLike
+    }
+    if (props.initialZoom) {
+      options.zoom = props.initialZoom
+    }
+  }
+
+  return options
+})
+
+onMounted(() => {
+
+  map.value = new Map(mapOptions.value)
+  map.value.on('load', () => {
+    map.value?.addSource('countries', {
+      type: 'geojson',
+      data: props.geoJsonData,
+    })
+
+    map.value?.addLayer({
+      id: 'countries-layer',
+      type: 'fill',
+      source: 'countries',
+      paint: layerPaint.value,
+    })
+
+    const popup = new Popup({
+      closeButton: false,
+      closeOnClick: false,
+    })
+
+    map.value?.on('mousemove', 'countries-layer', (e) => {
+      const feature = e.features?.[0]
+      if (feature) {
+        const { ISO_A2, ADMIN } = feature.properties
+        const metric = props.countryMetrics[ISO_A2]
+        if (metric !== undefined) {
+          popup.setLngLat(e.lngLat).setHTML(`<strong>${ADMIN}</strong>: ${metric} ${props.metricUnit}`).addTo(map.value as Map)
```

Comment:
non-blocking, just a thought. Since `setHTML` doesn't sanitize, hoping we can create a custom plugin here like we did for ChartJS.

Reply:
> non-blocking, just a thought. Since `setHTML` doesn't sanitize, hoping we can create a custom plugin here like we did for ChartJS.

This is maplibre's `setHTML` fn: https://maplibre.org/maplibre-gl-js/docs/API/classes/Popup/#sethtml 

We can followup on this later. values templated into the html (metric and metricUnit) are well typed and validated, so for simplicity we can just use maplibre's api for displaying a popup until we need something more "custom"